src/kriscv/kprovex/_default.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pathlib import Path
+    from typing import Final
+
+    from pyk.kast import KInner
+    from pyk.proof.reachability import APRProof
+
+    from .api import Config, Init, Show
+
+
+def init_from_claims(config: Config, spec_file: Path, claim_id: str) -> APRProof:
+    from pyk.ktool.claim_loader import ClaimLoader
+    from pyk.proof.reachability import APRProof
+
+    spec_module, claim_label = claim_id.split('.', 1)
+    include_dirs = config.dist.source_dirs
+
+    claims = ClaimLoader(config.kprove).load_claims(
+        spec_file=spec_file,
+        spec_module_name=spec_module,
+        claim_labels=[claim_label],
+        include_dirs=include_dirs,
+    )
+    (claim,) = claims
+
+    proof = APRProof.from_claim(
+        config.kprove.definition,
+        claim=claim,
+        logs={},
+        proof_dir=config.proof_dir,
+    )
+    return proof
+
+
+def show_pretty_term(config: Config, term: KInner) -> str:
```

Comment:
Similarly, all plugins support printing proof nodes using `kore-print`.

Reply:
Why are we using `kore_print`? I think we should use `Formatter` or `PrettyPrinter` instead, does `kore_print` corrcetly handle thinsg like variables?

Reply:
> does `kore_print` corrcetly handle thinsg like variables?

Yes, `kore_print` handles symbolic terms as well.

> Why are we using `kore_print`? I think we should use `Formatter` or `PrettyPrinter` instead

We could use `Formatter` for sure. An advantage of `kore_print` is that it handles an extra (non-trivial) case for parenthesization. Also, although I do not have measurements, for large terms, `kore_print` is probably significantly faster as it is implemented in C++.

Should we switch to `Formatter` as the default `Show` implementation?